src/hotspot/share/cds/filemap.hpp

```diff
@@ -587,11 +592,13 @@ class FileMapInfo : public CHeapObj<mtInternal> {
     return decode_start_address(spc, true);
   }
 
+public:
   // The starting address of spc, as calculated with HeapShared::decode_from_archive()
   address start_address_as_decoded_from_archive(FileMapRegion* spc) {
     return decode_start_address(spc, false);
   }
 
+private:
```

Comment:
Maybe move the public function to the public block before line 567 so that the labels at lines 595 and 601 could be removed.

Reply:
I think it's better to keep the functions together so you can see the difference between `start_address_as_decoded_with_current_oop_encoding_mode` and `start_address_as_decoded_from_archive`. I've shuffled the code a little to group the two related private methods together.